src/main/java/io/cdap/plugin/gcp/bigquery/sqlengine/BigQuerySQLEngine.java

```diff
@@ -506,6 +512,29 @@ public void cleanup(String datasetName) throws SQLEngineException {
     }
   }
 
+  /**
+   *
+   * @return capabilities provided by {@link BigQuerySQLEngine}, which include {@link StringExpressionFactoryType}.SQL.
+   */
+  @Override
+  public Set<Capability> getCapabilities() {
+    return Collections.singleton(StringExpressionFactoryType.SQL);
+  }
+
+  /**
+   *
+   * @return the single expression factory provided by {@link BigQuerySQLEngine}, which is {@link SQLExpressionFactory}.
+   */
+  @Override
+  public List<ExpressionFactory<?>> getExpressionFactories() {
+    return Collections.singletonList(new SQLExpressionFactory());
+  }
+
+  @Override
+  public Engine getRelationalEngine() {
+    return this;
+  }
+
```

Comment:
You also need `getRelation` and `transform`

Reply:
Added `getRelation`. `transform` will be added in a subsequent PR.

Reply:
Should it have ```aggregate``` and ```canAggregate``` (sorry I am following the design doc and it is  specified under "changes to BigQuerySQL" https://docs.google.com/document/d/1hYnP_jAday-RKI0TPkJX_r2iJeroZXZAZB8XWx1nTDI/edit?resourcekey=0-lS0CoG1Ji02whjnCwRZJ_w# ) or wondering if  it needs ```transform``` and ```canTransform```? (since it implements engine)

Reply:
`aggregate` and `canAggregate` will be added as helper methods which will be called from `transform` and `canTransform` respectively. In the future we can have `transform` and `canTransform` call methods such as `filter` and `canFilter`, `join` and `canJoin`, etc.

Reply:
got it. Thanks!